server/src/transport/http.rs

```diff
@@ -52,14 +53,22 @@ pub(crate) async fn reject_connection(socket: tokio::net::TcpStream) {
 
 /// The `Authority` can be sent by the client in the `Host header` or in the `URI`
 /// such that we must check both.
-pub(crate) fn fetch_authority(request: &hyper::Request<hyper::Body>) -> Option<&str> {
-	let host_header = http_helpers::read_header_value(request.headers(), hyper::header::HOST);
-	let uri = request.uri().authority();
+pub(crate) fn authority(request: &hyper::Request<hyper::Body>) -> Option<Authority> {
```

Comment:
Is this only used in the middleware now; I wonder whether it should move there?

Reply:
I moved it to `Authority::from_http_request` I think that's cleaner